redis/tests/test_auth.rs

```diff
@@ -0,0 +1,397 @@
+#![cfg(feature = "token-based-authentication")]
```

Comment:
these tests are all unit tests, and should be in the relevant files, not with external integration tests.

what you're missing is actual tests that check that connections get new credentials, and send these new credentials to the server.
You should create a test credentials provider, that doesn't rely on entra, and then either check that it refreshes the credentials with a mock server, or by spawning a task that periodically* changes the ACL password, and seeing that the connection is still usable.

*periodically here means every couple of milliseconds, because we don't want a test that runs for minutes or more.